src/runtime/Value.cpp

```diff
@@ -301,13 +301,15 @@ bool Value::abstractEqualsToSlowCase(ExecutionState& state, const Value& val) co
         double a = asNumber();
         double b = val.asNumber();
 
-        if (std::isnan(a) || std::isnan(b))
+        if (UNLIKELY(std::isnan(a) || std::isnan(b)))
             return false;
-        else if (a == b)
-            return true;
 
-        return false;
+        return a == b;
     } else {
+        if (u.asInt64 == val.u.asInt64) {
```

Comment:
Q. Can you explain why this is possible?

Reply:
If two operands have the same value, it means that they are equal.
So, return true directly without further checks below.